### ISSUE TITLE
DOTNET_CLI_UI_LANGUAGE not affecting the resource key TestRunningSummary

### ISSUE THREAD

**Issue by stefanforsberg** [microsoft/vstest#3029]
## Description
When setting the env variable DOTNET_CLI_UI_LANGUAGE the output from dotnet test is translated to the expected language except for the string represented by the TestRunningSummary which remains in English.

For reference, the TestRunningSummary resource I'm refering to is this one (for German): https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.Build/Resources/xlf/Resources.de.xlf#L15-L18

## Steps to reproduce

Set the environment variable DOTNET_CLI_UI_LANGUAGE to something other than your system setting(I'm running on EN as default). In this example I'm using powershell and setting: 
1: $Env:DOTNET_CLI_UI_LANGUAGE="de"
2: Execute the command dotnet test -t -v=q

## Expected behavior
```
Testlauf für "D:\path\to\some\project\nunitNet5\bin\Debug\net5.0\NunitTests.dll" (.NETCoreApp,Version=v5.0)
Microsoft (R) Testausführungs-Befehlszeilentool Version 16.8.0
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Die folgenden Tests sind verfügbar:
```

## Actual behavior
```
Test run for D:\path\to\some\project\nunitNet5\bin\Debug\net5.0\NunitTests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Testausführungs-Befehlszeilentool Version 16.8.0
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Die folgenden Tests sind verfügbar:
```

## Environment
Running on windows 10, 
dotnet sdk 5.0.100
<PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0" />

**Comment by Haplois** [comment]
Hi,

That resource was not localized in version [16.8.0][1]. I see that it works on 16.9 and upwards.


[1]: https://github.com/microsoft/vstest/blob/rel/16.8/src/Microsoft.TestPlatform.Build/Resources/xlf/Resources.de.xlf

**Comment by stefanforsberg** [comment]
I upgraded my sdk to v5.0.400 which seems to come with version 16.11.0 but the string is still printed in EN (eg disregards the DOTNET_CLI_UI_LANGUAGE setting) or am I missing something else here?

**Test run for** D:\path\to\some\project\nunitNet5\bin\Debug\net5.0\NunitTests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Testausführungs-Befehlszeilentool Version 16.11.0
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Die folgenden Tests sind verfügbar:

**Comment by Haplois** [comment]
Ah, sorry - I didn't isolate SDK installs. There are two issues here. The first issue was the unlocalized resource, and the second one is that TP didn't respect the value of that variable in our build task. The first issue is fixed in 16.9, and the second one is fixed in 17.0 previews. You can see the correct behavior in net6 preview runner.
